app/src/components/market/market_wizard_creator.tsx

```diff
@@ -38,97 +39,97 @@ interface State {
   marketData: MarketData
 }
 
-export class MarketWizardCreator extends Component<Props, State> {
-  public state: State = {
-    currentStep: 1,
-    marketData: {
-      collateralId: 'dai',
-      question: '',
-      category: '',
-      resolution: null,
-      arbitratorId: 'realitio',
-      spread: '1',
-      funding: new BigNumber('0'),
-      outcomes: [
-        {
-          name: 'Yes',
-          probability: 50,
-        },
-        {
-          name: 'No',
-          probability: 50,
-        },
-      ],
-    },
+export const MarketWizardCreator = (props: Props) => {
+  const { callback, status, questionId, marketMakerAddress } = props
+
+  const marketDataDefault: MarketData = {
+    collateralId: 'dai',
+    collateralsCustom: [],
+    question: '',
+    category: '',
+    resolution: null,
+    arbitratorId: 'realitio',
+    spread: '1',
+    funding: new BigNumber('0'),
+    outcomes: [
+      {
+        name: 'Yes',
+        probability: 50,
+      },
+      {
+        name: 'No',
+        probability: 50,
+      },
+    ],
+  }
+
+  const [currentStep, setCurrentStep] = useState(1)
+  const [marketData, setMarketdata] = useState<MarketData>(marketDataDefault)
+
+  const next = (): void => {
+    const actualCurrentStep = currentStep >= 3 ? 4 : currentStep + 1
+    setCurrentStep(actualCurrentStep)
   }
 
-  public next = (): void => {
-    let { currentStep } = this.state
-    currentStep = currentStep >= 3 ? 4 : currentStep + 1
-    this.setState({
-      currentStep,
-    })
+  const back = (): void => {
+    const actualCurrentStep = currentStep <= 1 ? 1 : currentStep - 1
+    setCurrentStep(actualCurrentStep)
   }
 
-  public back = (): void => {
-    let { currentStep } = this.state
-    currentStep = currentStep <= 1 ? 1 : currentStep - 1
-    this.setState({
-      currentStep,
-    })
+  const addCollateralCustom = (collateral: Collateral): void => {
+    const collateralsCustom = marketData.collateralsCustom
+    collateralsCustom.push(collateral)
+    const newMarketData = {
+      ...marketData,
+      collateralsCustom,
+    }
+    setMarketdata(newMarketData)
   }
 
-  public handleChange = (
-    event: ChangeEvent<HTMLInputElement> | ChangeEvent<HTMLSelectElement> | BigNumberInputReturn,
+  const handleChange = (
+    event:
+      | ChangeEvent<HTMLInputElement>
+      | ChangeEvent<HTMLSelectElement>
+      | BigNumberInputReturn
+      | CollateralCustomEvent,
```

Comment:
This is getting out of hand IMO. I would just use a custom event handler like we do below with `handleOutcomesChange` and `handleDateChange`.